ros2doctor/ros2doctor/verb/call.py

```diff
@@ -0,0 +1,260 @@
+
+
+class Talker(Node):
+    """Initialize talker node."""
+
+    def __init__(self, topic, time_period, *, qos=10):
+        super().__init__('ros2doctor_talker')
+        self.i = 0
+        self.pub = self.create_publisher(String, topic, qos)
+        self.timer = self.create_timer(time_period, self.timer_callback)
+
+    def timer_callback(self):
+        msg = String()
+        hostname = socket.gethostname()
+        # publish
+        msg.data = f"hello, it's me {hostname}"
+        summary_table.increment_pub()
+        self.pub.publish(msg)
+        self.i += 1
+
+
+class Listener(Node):
+    """Initialize listener node."""
+
+    def __init__(self, topic, *, qos=10):
+        super().__init__('ros2doctor_listener')
+        self.sub = self.create_subscription(
+            String,
+            topic,
+            self.sub_callback,
+            qos)
+
+    def sub_callback(self, msg):
+        # subscribe
+        msg_data = msg.data.split()
+        pub_hostname = msg_data[-1]
+        if pub_hostname != socket.gethostname():
+            summary_table.increment_sub(pub_hostname)
+
+
+def _send(*, group=DEFAULT_GROUP, port=DEFAULT_PORT, ttl=None):
+    """Multicast send one message."""
+    hostname = socket.gethostname()
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+    if ttl is not None:
+        packed_ttl = struct.pack('b', ttl)
+        s.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_TTL, packed_ttl)
+    try:
+        s.sendto(f"hello, it's me {hostname}".encode('utf-8'), (group, port))
+        summary_table.increment_send()
+    finally:
+        s.close()
+
+
+def _receive(*, group=DEFAULT_GROUP, port=DEFAULT_PORT, timeout=None):
+    """Multicast receive."""
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+    try:
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        try:
+            s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
+        except AttributeError:
+            # not available on Windows
+            pass
+        s.bind(('', port))
+
+        s.settimeout(timeout)
+
+        mreq = struct.pack('4sl', socket.inet_aton(group), socket.INADDR_ANY)
+        s.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, mreq)
+        try:
+            data, _ = s.recvfrom(4096)
+            data = data.decode('utf-8')
+            sender_hostname = data.split()[-1]
+            if sender_hostname != socket.gethostname():
+                summary_table.increment_receive(sender_hostname)
+        finally:
+            s.setsockopt(socket.IPPROTO_IP, socket.IP_DROP_MEMBERSHIP, mreq)
+    finally:
+        s.close()
+
+
+class SummaryTable():
+    """Summarize number of msgs published/sent and subscribed/received."""
+
+    def __init__(self):
+        """Initialize empty summary table."""
+        self.lock = threading.Lock()
+        self.pub = 0
+        self.send = 0
+        self.sub = {}
+        self.receive = {}
```

Comment:
nitpick: I think all these variables can be prefixed with an underscore since they are implementation details.